src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMethodSymbolComparer.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using Acuminator.Utilities.Common;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PXOverrideMismatch
+{
+	public class PXOverrideMethodSymbolComparer : IEqualityComparer<IMethodSymbol>
```

Comment:
I don't recommend to move out the comparison logic out of the analyzer. It's not wrong to have a separate component for it but in the future we will most likely enhance the checks and report different cases of partial compatibility between the method, so there will be extra analysis logic added in the future which won't fit well into `IEqualityComparer` contract.